crates/pop-cli/src/commands/bench/overhead.rs

```diff
@@ -0,0 +1,346 @@
+
+		// Prompt user to update output path of the benchmarking results.
+		if cmd.params.weight.weight_path.is_none() {
+			let input = cli
+				.input("Provide the output directory path for weight files")
+				.required(true)
+				.placeholder(".")
+				.default_input(".")
+				.interact()?;
+			cmd.params.weight.weight_path =
+				if !input.is_empty() { Some(input.into()) } else { None };
+		}
+		Ok(())
+	}
+
+	async fn run(&self, cli: &mut impl cli::traits::Cli) -> anyhow::Result<()> {
+		let binary_path = check_omni_bencher_and_prompt(cli, self.skip_confirm).await?;
+		generate_omni_bencher_benchmarks(
+			binary_path.as_path(),
+			"overhead",
+			self.collect_arguments(),
+			false,
+		)?;
+		Ok(())
+	}
+
+	fn display(&self) -> String {
+		let mut args = vec!["pop bench overhead".to_string()];
+		let mut arguments = self.collect_arguments();
+		if let Some(ref profile) = self.profile {
+			arguments.push(format!("--profile={}", profile));
+		}
+		if self.skip_confirm {
+			arguments.push("--skip-confirm".to_string());
+		}
+		args.extend(arguments);
+		args.join(" ")
+	}
+
+	fn collect_arguments(&self) -> Vec<String> {
+		let mut arguments: Vec<String> = std::env::args()
+			.skip(3)
+			// Exclude custom arguments which are not in the `OverheadCommand`.
+			.filter(|arg| !EXCLUDED_ARGS.iter().any(|a| arg.starts_with(a)))
+			.collect();
+
+		// Check if the arguments are provided by the user.
+		let mut print_runtime = true;
+		let mut print_genesis_builder = true;
+		let mut print_genesis_builder_preset = true;
+		let mut print_weight_path = true;
+		for argument in arguments.iter() {
+			print_runtime = print_runtime && !argument.starts_with("--runtime");
+			print_genesis_builder =
+				print_genesis_builder && !argument.starts_with("--genesis-builder");
+			print_genesis_builder_preset =
+				print_genesis_builder_preset && !argument.starts_with("--genesis-builder-preset");
+			print_weight_path = print_weight_path && !argument.starts_with("--weight-path");
+		}
+
+		if print_runtime {
+			if let Some(ref runtime) = self.command.params.runtime {
+				arguments.push(format!("--runtime={}", runtime.display()));
+			}
+		}
+		if print_genesis_builder {
+			arguments.push("--genesis-builder=runtime".to_string());
+		}
+		if print_genesis_builder_preset {
+			arguments.push(format!(
+				"--genesis-builder-preset={}",
+				self.command.params.genesis_builder_preset
+			));
+		}
+		if print_weight_path {
+			if let Some(ref weight_path) = self.command.params.weight.weight_path {
+				arguments.push(format!("--weight-path={}", weight_path.display()));
+			}
+		}
+		arguments
+	}
+}
+
+fn parse_genesis_builder_policy(policy: &str) -> anyhow::Result<OverheadCmd> {
+	OverheadCmd::try_parse_from([
+		"",
+		"--runtime",
+		"dummy-runtime", // For parsing purpose.
+		"--genesis-builder",
+		policy,
+	])
+	.map_err(|e| {
+		anyhow::anyhow!(format!(r#"Invalid genesis builder option {policy}: {}"#, e.to_string()))
+	})
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
There is a lot of `unwrap()`, I guess in tests is ok but because returning `anyhow::Result<()>` we probably can use `?` too